tpl/tplimpl/shortcodes_integration_test.go

```diff
@@ -0,0 +1,328 @@
+// Copyright 2025 The Hugo Authors. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package tplimpl_test
+
+import (
+	"strings"
+	"testing"
+
+	"github.com/gohugoio/hugo/htesting/hqt"
+	"github.com/gohugoio/hugo/hugolib"
+)
+
+func TestCommentShortcode(t *testing.T) {
+	t.Parallel()
+
+	files := `
+-- hugo.toml --
+disableKinds = ['page','rss','section','sitemap','taxonomy','term']
+-- layouts/index.html --
+{{ .Content }}
+-- content/_index.md --
+a{{< comment >}}b{{< /comment >}}c
+`
+
+	b := hugolib.Test(t, files)
+	b.AssertFileContent("public/index.html", "<p>ac</p>")
+}
+
+func TestDetailsShortcode(t *testing.T) {
+	t.Parallel()
+
+	files := `
+-- hugo.toml --
+disableKinds = ['page','rss','section','sitemap','taxonomy','term']
+-- layouts/index.html --
+{{ .Content }}
+-- content/_index.md --
+---
+title: home
+---
+{{< details >}}
+A: An _emphasized_ word.
+{{< /details >}}
+
+{{< details
+  class="my-class"
+  name="my-name"
+  open=true
+  summary="A **bold** word"
+  title="my-title"
+>}}
+B: An _emphasized_ word.
+{{< /details >}}
+
+{{< details open=false >}}
+C: An _emphasized_ word.
+{{< /details >}}
+
+{{< details open="false" >}}
+D: An _emphasized_ word.
+{{< /details >}}
+
+{{< details open=0 >}}
+E: An _emphasized_ word.
+{{< /details >}}
+`
+	b := hugolib.Test(t, files)
+
+	b.AssertFileContent("public/index.html",
+		"<details>\n  <summary>Details</summary>\n  <p>A: An <em>emphasized</em> word.</p>\n</details>",
+		"<details class=\"my-class\" name=\"my-name\" open title=\"my-title\">\n  <summary>A <strong>bold</strong> word</summary>\n  <p>B: An <em>emphasized</em> word.</p>\n</details>",
+		"<details>\n  <summary>Details</summary>\n  <p>C: An <em>emphasized</em> word.</p>\n</details>",
+		"<details>\n  <summary>Details</summary>\n  <p>D: An <em>emphasized</em> word.</p>\n</details>",
+		"<details>\n  <summary>Details</summary>\n  <p>D: An <em>emphasized</em> word.</p>\n</details>",
+	)
+}
+
+func TestInstagramShortcode(t *testing.T) {
+	t.Parallel()
+
+	files := `
+-- hugo.toml --
+disableKinds = ['page','rss','section','sitemap','taxonomy','term']
+privacy.instagram.simple = false
+-- content/_index.md --
+---
+title: home
+---
+{{< instagram CxOWiQNP2MO >}}
+-- layouts/index.html --
+{{- .Content | hash.XxHash -}}
+`
+
+	// Regular mode
+	b := hugolib.Test(t, files)
+	b.AssertFileContentEquals("public/index.html", "a7937c49665872d3")
```

Comment:
So, I like the concept of using hashes here, but I suspect I'm going to get bitten by the lack of output at some point. 

I suggest that we print both the hash and the content, e.g.

```
Hash: {{ .Content | hash.XxHash }}
Content: {{ .Content }}
```

And then do:

```
b.AssertFileContent("public/index.html", "a7937c49665872d3")
```

The test will be as robust as before, but we will get better output on failures.